test/unit/org/apache/cassandra/repair/autorepair/FixedSplitTokenRangeSplitterVNodesTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.repair.autorepair;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Unit tests for a setup that has v-nodes {@link FixedSplitTokenRangeSplitter}
+ */
+@RunWith(Parameterized.class)
+public class FixedSplitTokenRangeSplitterVNodesTest
+{
+    private static final int numTokens = 16;
+
+    @Parameterized.Parameter(0)
+    public AutoRepairConfig.RepairType repairType;
+
+    @Parameterized.Parameter(1)
+    public int numberOfSubRanges;
+
+    @Parameterized.Parameters(name = "repairType={0}, numberOfSubRanges={1}")
+    public static Collection<Object[]> parameters()
+    {
+        List<Object[]> params = new ArrayList<>();
+        for (AutoRepairConfig.RepairType type : AutoRepairConfig.RepairType.values())
+        {
+            for (int subRange : Arrays.asList(1, 2, 4, 8, 16, 32, 64, 128, 256))
```

Comment:
One thing that could be interesting i just realized is if the subranges doesn't divide vnodes evenly, you could get value < number of subranges.  e.g.   with 16 `num_tokens` and 20 `number_of_subranges`  i'd expect it only produces `16` tokens.   I think this is fine and don't think we should attempt to non-evenly subdivide between the ranges.

Reply:
Yeah, I think we can ignore it for now.